### PR TITLE
Major upgrade to fully support webpack 4.

### PR DESCRIPTION
**Pull request by jcgregorio**
Use the module.exports = (env, argv) => {...} form that webpack supports to capture the mode,
either production or development. See:

  https://github.com/webpack/webpack/issues/6460#issuecomment-386947990

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 node_modules
 dist
-example/package-lock.json
\ No newline at end of file
+example/package-lock.json
+package-lock.json
```
Comment (kjlubick) [inline .gitignore:4]: I'm confused, don't we want to check in the package-lock.json?

File: README.md [hunk 2]
```diff
@@ -127,9 +127,12 @@ Sometimes, an app wants to specify that the js/css files are in an absolute path
 Since Pulito just returns a Webpack object, the output of `commonBuilder` can be
 modified directly, like:
 
-    const { commonBuilder } = require('pulito');
-    module.exports = commonBuilder(__dirname);
-    module.exports.output.publicPath='/static/';
+    const commonBuilder = require('pulito');
+    module.exports = (env, argv) => {
```
Comment (kjlubick) [inline README.md:131]: I'm doubly confused - module.exports is an object OR a function that returns an object OR?  Is that new to webpack 4?

File: example/package-lock.json [hunk 3]
```diff
@@ -1,9 +1,36 @@
 {
```
Comment (kjlubick) [inline example/package-lock.json:1]: I think we still don't want to check-in/zip the example/package-lock.json, right?

File: example/webpack.config.js [hunk 4]
```diff
@@ -1,2 +1,4 @@
-const { commonBuilder } = require('pulito');
-module.exports = commonBuilder(__dirname);
+const commonBuilder = require('pulito');
+//const commonBuilder = require('../webpack.common.js');
```
Comment (kjlubick) [inline example/webpack.config.js:2]: Remove comment

File: .gitignore (hunk 1 above)
Comment (jcgregorio) [reply .gitignore:4]: No until we start using it, package-lock.json is only used when you run "npm ci", which we don't do. Maybe if we start using TravisCI we can put it back.

File: README.md (hunk 2 above)
Comment (jcgregorio) [reply README.md:131]: Yes, it can be either obj or function. Not sure if that's new in webpack 4.

File: example/package-lock.json (hunk 3 above)
Comment (jcgregorio) [reply example/package-lock.json:1]: Right, I missed that, I think fixed when I pushed after the 4.0 release.

**Pull request merged**